pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -853,6 +860,8 @@ func saveHostProvisioningSettings(host *metal3v1alpha1.BareMetalHost) (dirty boo
 		dirty = true
 	}
 
+	dirty = updateBootModeStatus(host)
```

Comment:
This can overwrite a change to the root device hints. Should be:

```
if updateBootModeStatus(host) {
    dirty = true
}
```

or perhaps

```
dirty |= updateBootModeStatus(host)
```

Reply:
Done